ctapipe/analysis/sensitivity.py

```diff
@@ -0,0 +1,733 @@
+
+    return rates
+
+
+def sigma_lima(Non, Noff, alpha=0.2):
+    """
+    Compute the significance according to Eq. (17) of Li & Ma (1983).
+
+    Parameters
+    ----------
+    Non : integer
+        Number of on counts
+    Noff : integer
+        Number of off counts
+    alpha : float, optional (default: 0.2)
+        Ratio of on-to-off exposure
+
+    Returns
+    -------
+    sigma : float
+        the significance of the given off and on counts
+    """
+
+    alpha1 = alpha + 1.0
+    sum = Non + Noff
+    arg1 = Non / sum
+    arg2 = Noff / sum
+    term1 = Non * np.log((alpha1/alpha)*arg1)
+    if Noff == 0:
+        term2 = 0
+    else:
+        term2 = Noff * np.log(alpha1*arg2)
+    sigma = np.sqrt(2.0 * (term1 + term2))
+
+    return sigma
+
+
+def diff_to_X_sigma(scale, N, alpha, X=5):
+    """
+    calculates the significance according to `sigma_lima` and returns the squared
+    difference to X. To be used in a minimiser that determines the necessary source
+    intensity for a detection of given significance of `X` sigma
+    The square here is only to have a continuously differentiable function with a smooth
+    turning point at the minimum -- in contrast to an absolute-function that makes a
+    sharp turn at the minimum.
+
+    Parameters
+    ----------
+    scale : python list with a single float
+        this is the variable in the minimisation procedure
+        it scales the number of gamma events
+    N : shape (2) list
+        the signal count in the on- (index 0)
+        and the background count int the off-region (index 1)
+        the events in the on-region are to be scaled by `scale[0]`
+        the background rate in the on-region is estimated as `alpha` times off-count
+    alpha : float
+        the ratio of the on and off areas
+    X : float, optional (default: 5)
+        target significance in multiples of "sigma"
+
+    Returns
+    -------
+    (sigma-X)**2 : float
+        squared difference of the significance to `X`
+        minimise this function for the number of gamma events needed for your desired
+        significance of `X` sigma
+    """
+
+    Non = N[1]*alpha + N[0] * scale[0]
+    Noff = N[1]
+    sigma = sigma_lima(Non, Noff, alpha)
+    return (sigma-X)**2
+
+
+#   ######  ##          ###     ######   ######
+#  ##    ## ##         ## ##   ##    ## ##    ##
+#  ##       ##        ##   ##  ##       ##
+#  ##       ##       ##     ##  ######   ######
+#  ##       ##       #########       ##       ##
+#  ##    ## ##       ##     ## ##    ## ##    ##
+#   ######  ######## ##     ##  ######   ######
+class SensitivityPointSource():
+    """
+    class to calculate the sensitivity to a known point-source
+    TODO extend sensitvity module
+        • add extended source?
+        • add pseudo experiment for low exposure times?
+    """
+    def __init__(self, mc_energies=None, energy_bin_edges=None,
+                 energy_unit=u.TeV, flux_unit=1/(u.TeV*u.m**2*u.s)):
+        """
```

Comment:
At least in Astropy and Gammapy, docstrings are always class-level and the `__init__` docstring is left empty.
I suspect (but didn't check) that the same pattern is used in ctapipe.
So suggest to move the content of this docstring to the class-level docstring.